Engine/platform/base/sys_main.cpp

```diff
@@ -193,6 +193,7 @@ SDL_Window *sys_get_window() {
 
 void sys_window_set_style(WindowMode mode, Size size) {
     if (!window) return;
+    if(mode != kWnd_Windowed && !platform->FullscreenSupported()) mode = kWnd_Windowed;
```

Comment:
Only one small nitpick: please use our codestyle, I noticed you are sometimes writing the new code using this different style:
* there has to be a space after `if`
* please place the statements under `if` on the next line; I know there have been several cases added where it's on the same line (either by nick sonneveld, or by me following his example), but these were the return statements, and here it's an assignment, also it becomes especially inconvenient to read the code when there's a long condition like this.